pom.xml

```diff
@@ -167,6 +167,12 @@
       <version>3.0</version>
     </dependency>
 
+    <dependency>
```

Comment:
This is an unexected dependency to me. Looking further in the patch I start to understand it: Aether has Nodes, whereas maven-artifact-transfer has not.
Makes we wonder if we should merge the 2 projects into one.

Reply:
May be. m-dependency-tree has only 1 project functionality (that is retrieve the resolved dependency tree), but many Node related functionality (the node class, filters, and traversal visitors) that I needed. Another option is to create a new shared lib with that node only functionality (which actually does not depend on Aether nor Maven versions).

Reply:
The whole idea is that maven-artifact-transfer is an independend layer around Aether/Maven Artifact Resolver. It act as a bridge to the different implementations (well... more or less different packages). Hence our response when we notices implementations for different Maven versions.
I don't think we need another library to do so.
@khmarbaise WDYT?

Reply:
I agree on that, cause that is the basic idea of maven-artifact-transfer....not another library..
And yes it looks to me we need to think about merging the node part from m-dependency-tree...

It might be the case that some functionality is not yet in maven-artifact-transfer so we simple need to enhance it...cause it has shown already that it makes several usage in plugins etc. and also in extensions much easier and more convenient...

Reply:
+1 to merge the node functionality...will it be in another issue? or should I add a commit on this PR?
Also, please review the ITs, specially there is one on optional depencencies that differs in result in Maven 3.0.x and 3.1+. The ITs I wrote were those cases I could think from the top of my head but surely there are more cases to test.

Reply:
Should I start the merge of m-dependency-tree to m-artifact-transfer?